prow/test/integration/test/webhook_server_test.go

```diff
@@ -55,11 +67,64 @@ func TestWebhookServerValidateCluster(t *testing.T) {
 	//prowjob should be created successfully
 	err = exec.Command("kubectl", "--context=kind-kind-prow-integration", "apply", "-f", absValidProwJobPath).Run()
 	if err != nil {
-		t.Errorf("prowjob was not properly validated")
+		t.Errorf("prowjob was not properly validated %v", err)
 	}
 	//cleanup prowjob that was created
 	err = cleanUpCluster(validProwJobID)
 	if err != nil {
 		t.Errorf("could not delete prowjob %s", validProwJobID)
 	}
 }
+
+func TestWebhookServerMutateProwjob(t *testing.T) {
+	validBareProwJobPath := "./testdata/valid_bare_prowjob.yaml"
+	absValidBareProwJobPath, err := filepath.Abs(validBareProwJobPath)
+	var mutatedProwJob v1.ProwJob
+	if err != nil {
+		t.Fatalf("could not find absolute file path")
+	}
+	//prowjob should be created successfully
+	err = exec.Command("kubectl", "--context=kind-kind-prow-integration", "apply", "-f", absValidBareProwJobPath).Run()
+	if err != nil {
+		t.Errorf("prowjob was not properly validated %v", err)
+	}
+
+	key := types.NamespacedName{
+		Namespace: "default",
+		Name:      validBareProwJobID,
+	}
+	cfg := config.GetConfigOrDie()
+	if err != nil {
+		t.Fatalf("Error getting kubeconfig")
+	}
+	client, err := ctrlruntimeclient.New(cfg, ctrlruntimeclient.Options{})
+	if err != nil {
+		t.Fatalf("Could not create writer client %v", err)
+	}
+	ctx := context.Background()
+	if err := client.Get(ctx, key, &mutatedProwJob); err != nil {
+		t.Errorf("could not get prowjob: %v", err)
+	}
+
+	originalProwJobYAML, err := ioutil.ReadFile(absValidBareProwJobPath)
+	if err != nil {
+		t.Fatalf("unable to read yaml file %v", err)
+	}
+	var originalProwJob v1.ProwJob
+	err = yaml.Unmarshal(originalProwJobYAML, &originalProwJob)
+	if err != nil {
+		t.Fatalf("unable to unmarshal into prowjob %v", err)
+	}
+	
+	diff := cmp.Diff(originalProwJob, mutatedProwJob)
+	if diff == "" {
+		t.Errorf("prowjob was not mutated %v", err)
+	}
+
+	//cleanup prowjob that was created
+	err = cleanUpCluster(validBareProwJobID)
```

Comment:
Please define `t.Cleanup` very early on in this test, which behaves like `finally` in Java